src/cargo/core/compiler/mod.rs

```diff
@@ -76,6 +75,22 @@ pub struct BuildConfig {
     pub json_messages: bool,
 }
 
+impl BuildConfig {
+    pub fn new(host_triple: &str, requested_target: &Option<String>) -> CargoResult<BuildConfig> {
+        if let Some(ref s) = *requested_target {
+            if s.trim().is_empty() {
+                bail!("target was empty")
+            }
+        }
+        Ok(BuildConfig {
+            host_triple: host_triple.to_string(),
+            requested_target: (*requested_target).clone(),
+            jobs: 1,
```

Comment:
Let's pass in `jobs: Option<u32>`?

Reply:
The reason I didn't do that is that passing `None` explicitly always feels mostly ugly, especially in this situation where there are only two callers, and `1` seems pretty defensible as a default. All the fields of the struct are already being modified, anyway.

Reply:
(But if you insist, sure!)